data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateProcessor.java

```diff
@@ -85,11 +84,14 @@ public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
 
         final List<Map.Entry<AggregateIdentificationKeysHasher.IdentificationKeysMap, AggregateGroup>> groupsToConclude = aggregateGroupManager.getGroupsToConclude(forceConclude);
         for (final Map.Entry<AggregateIdentificationKeysHasher.IdentificationKeysMap, AggregateGroup> groupEntry : groupsToConclude) {
-            final Optional<Event> concludeGroupEvent = aggregateActionSynchronizer.concludeGroup(groupEntry.getKey(), groupEntry.getValue(), forceConclude);
-
-            if (concludeGroupEvent.isPresent()) {
-                recordsOut.add(new Record(concludeGroupEvent.get()));
-                actionConcludeGroupEventsOutCounter.increment();
+            final AggregateActionOutput actionOutput = aggregateActionSynchronizer.concludeGroup(groupEntry.getKey(), groupEntry.getValue(), forceConclude);
+
+            final List<Event> concludeGroupEvents = actionOutput != null ? actionOutput.getEvents() : null;
+            if (concludeGroupEvents != null && !concludeGroupEvents.isEmpty()) {
```

Comment:
Nit: Would be nice if `AggregateActionOutput` was never null, and always returned an object with at least an empty list of events to not have to do all these null checks